### PR TITLE
Fixed broken patient view page table download

### PR DESCRIPTION
**Pull request by onursumer**
# What? Why?
Fix https://github.com/cBioPortal/cbioportal/issues/3563.

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

### CODE REVIEW

**Comment by onursumer** [comment]
I think we definitely needed to add tests for this fairly complex logic of "breaking single rows into multiple rows when downloading"...

File: src/pages/patientView/mutation/column/AlleleFreqColumnFormatter.tsx [hunk 1]
```diff
@@ -161,18 +153,37 @@ export default class AlleleFreqColumnFormatter {
 
 
     public static getFrequency(data:Mutation[]): string|string[] {
-        let result = [];
+        const result: string[] = [];
+
         if (data) {
-            for (let mutation of data) {
-                let frequency = mutation.tumorAltCount/(mutation.tumorAltCount+mutation.tumorRefCount);
-                if (frequency) {
```
Comment (onursumer) [inline src/pages/patientView/mutation/column/AlleleFreqColumnFormatter.tsx]: This was filtering out zero values like `0.0`, that's why we were ending up with an smaller array size and getting the error message `Not all the arrays returned from the download functions are from the same length.`

File: src/pages/patientView/mutation/column/AlleleFreqColumnFormatter.tsx [hunk 2]
```diff
@@ -161,18 +153,37 @@ export default class AlleleFreqColumnFormatter {
 
 
     public static getFrequency(data:Mutation[]): string|string[] {
```
Comment (onursumer) [inline src/pages/patientView/mutation/column/AlleleFreqColumnFormatter.tsx]: Should we name this function `download` for consistency?

Comment (oplantalech) [reply src/pages/patientView/mutation/column/AlleleFreqColumnFormatter.tsx]: No objections, go ahead to change the name if it is more consistent with the code.

**Comment by oplantalech** [comment]
@onursumer I've tested your code with the sample I put as example in my PR ([link here](http://www.cbioportal.org/beta/case.do?cancer_study_id=lgg_ucsf_2014&case_id=P04#/patient?caseId=P04&studyId=lgg_ucsf_2014)) and some allelic frequencies now appear empty:
captura de pantalla 2018-01-08 a les 16 08 03

**Comment by onursumer** [comment]
@oplantalech an empty value indicates that there is no allele frequency data for that specific sample. For example, the first row with empty allele frequency value in your example (the one starting with the protein change `N49Tfs*23`) does not have any allele frequency value. See the screenshot below (taken from the example link you shared).

empty_allele_frequency

Current download logic generates a file with an allele frequency value of `1.05882352941176` for the same column though. Any idea where that value is coming from?

**Comment by oplantalech** [comment]
@onursumer good you discovered this! It is a bug of course, I don't have any idea how this can happen, if I have time I'll look it up and post it here. Meanwhile, I approve the PR.

**Comment by pieterlukasse** [comment]
@oplantalech did you find the bug?

**Pull request merged**